src/main/java/org/checkerframework/checker/noliteral/qual/NonConstant.java

```diff
@@ -0,0 +1,29 @@
+package org.checkerframework.checker.noliteral.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.DefaultFor;
+import org.checkerframework.framework.qual.DefaultInUncheckedCodeFor;
+import org.checkerframework.framework.qual.DefaultQualifierInHierarchy;
+import org.checkerframework.framework.qual.SubtypeOf;
+import org.checkerframework.framework.qual.TypeUseLocation;
+
+/**
+ * The bottom type for the NoLiteral Checker's type system.
```

Comment:
This documentation never says what `@NonConstant` means, which is the most important thing it should say.